mlx/ops.cpp

```diff
@@ -753,6 +753,43 @@ array repeat(const array& arr, int repeats, StreamOrDevice s) {
   return repeat(flatten(arr, s), repeats, 0, s);
 }
 
+/** Replicating array a specified number of times along each axis */
+array tile(const array& a, std::vector<int> repeats, StreamOrDevice s) {
+  if (repeats.size() == 0) {
+    return a;
+  }
+
+  for (size_t i = 0; i < repeats.size(); i++) {
+    if (repeats[i] < 0) {
+      throw std::invalid_argument(
+          "[tile] Negative repeat value found at index " + std::to_string(i));
```

Comment:
Extremely nitpicky, feel free to ignore. Using an `ostringstream` would be better as it would allow you to also provide the whole repeats vector. Something like:

```
std::ostringstream msg;
msg << "[tile] Repeats argument cannot contain negative values but "
    << repeats << " was provided.";
throw std::invalid_argument(msg.str());
```